AmplifyPlugins/DataStore/AWSDataStoreCategoryPlugin/Sync/SubscriptionSync/ReconcileAndLocalSave/ReconcileAndLocalSaveOperation.swift

```diff
@@ -278,7 +297,7 @@ class ReconcileAndLocalSaveOperation: AsynchronousOperation {
         let version = savedModel.syncMetadata.version
         if savedModel.syncMetadata.deleted {
             mutationType = .delete
-        } else if version == 1 {
+        } else if version == 1 || existsLocally {
```

Comment:
if you never had the model in local store, and you apply the remote model to local store then save the mutationSync metadata with version 1, you'll take the syncMetadata and add it to the appliedModel before passing it to `notify`. if applied model was version=1 because someone else created it and it got synchronized to this local store, how do we know it was a create? 

I thought this was a bug so that you added the query against syncMetadata before saving the new remote sync metadata. if that's the case, you should remove the check for version 1 completely 

```
        } else if existsLocally {
             mutationType = .update 
        } else {  
        mutationType = .create
       }   
```
```

Reply:
or i may be missing something here that requires us to check version 1 first?

Reply:
if the version comes from source of truth is 1, that must be `.create`, right? and the following `!existsLocally` implicitly saying if `version > 1 && !existsLocally`, then it is also type of `.create`, right?